asynch/connection.py

```diff
@@ -15,38 +15,46 @@ def __init__(
         user: str = constants.DEFAULT_USER,
         password: str = constants.DEFAULT_PASSWORD,
         host: str = constants.DEFAULT_HOST,
-        port: int = constants.DEFAULT_PORT,
+        port: int | None = None,
         database: str = constants.DEFAULT_DATABASE,
         cursor_cls=Cursor,
         echo: bool = False,
         stack_track: bool = False,
+        secure: bool = False,
         **kwargs,
     ):
+        if secure and not port:
+            port = constants.DEFAULT_SECURE_PORT
+        elif not secure and not port:
+            port = constants.DEFAULT_PORT
         if dsn:
             config = parse_dsn(dsn)
-            self._connection = ProtoConnection(**config, stack_track=stack_track, **kwargs)
             user = config.get("user", None) or user
             password = config.get("password", None) or password
             host = config.get("host", None) or host
             port = config.get("port", None) or port
             database = config.get("database", None) or database
+            secure = config.get("secure", None) or secure
```

Comment:
```suggestion
secure = config.get("secure") or secure
```

And the same for the above statements if you like.